### ISSUE TITLE
A better name for the ``"derivative"`` method of error propagation?

### ISSUE THREAD

**Issue by astralcai** [Queens-Physics/qexpy#61]
Currently, the standard method of error propagation is called the ``"derivative"`` method, which seems quite strange. I think this name was carried over from the original version of qexpy, before my first refactor. 

I've been working on another refactor, and I'm wondering if I should change the name. I've seen it called the ``"quadrature"`` method more often, or even the ``"variance"`` method. Is there an actual formal name for it?

If you're worried about backwards compatibility, we can still keep the old name for a year, but giving a deprecated warning to users whenever they use the old name.

Let me know what you think. @ryanmartinneutrino

**Comment by CKapahi** [comment]
Something to think about, the textbook that @ryanmartinneutrino wrote (at least the version I have) refers to this method as "Adding in quadrature" or "Uncertainty in a function of ...". It will probably be worth standardizing terminology between the course material and QExPy.

**Comment by ryanmartinneutrino** [comment]
I'm really not sure on the best name to use! It's not just quadrature if the covariance term is included, and somehow it should refer to the fact that one is taking partial derivatives of a multi-variate function analytically (and dropping higher order derivatives). "derivative" method seems like the easiest to understand, but I'm fine with other suggestions.

**Comment by astralcai** [comment]
> I'm really not sure on the best name to use! It's not just quadrature if the covariance term is included, and somehow it should refer to the fact that one is taking partial derivatives of a multi-variate function analytically (and dropping higher order derivatives). "derivative" method seems like the easiest to understand, but I'm fine with other suggestions.

That is a good point. I thought ``"quadrature"`` would be better because it is the term used in the textbook as Connor mentioned. I just don't really like the name ``"derivative"`` because it sounds quite informal, but I've been looking through papers to find a more "official" name for this method, and there really isn't one. This method is typically just referred to as "the error propagation formula", since it is the only valid analytical method that is widely used, so I guess people never felt the need to coming up with a name for it.

One website as well as another paper I've found calls it the [moment method](http://www.cs.ox.ac.uk/devendra.ghate/litProp.html), which seems more mathematically accurate but even more confusing for users than ``"derivative"``. Another alternative I've found is calling it the ``"variance"`` method, but this isn't really informative about how the method works as you pointed out. In the absence of a better alternative, I'm ok with going back to ``"derivative"`` if you feel that's more appropriate. 

Let me know what you think. @ryanmartinneutrino
